common/djangoapps/student/roles.py

```diff
@@ -6,11 +6,33 @@
 from abc import ABCMeta, abstractmethod
 
 from django.contrib.auth.models import User
+import logging
 
 from student.models import CourseAccessRole
 from xmodule_django.models import CourseKeyField
 
 
+log = logging.getLogger(__name__)
+
+# A list of registered access roles.
+REGISTERED_ACCESS_ROLES = {}
+
+
+def register_access_role(cls):
+    """
+    Decorator that allows access roles to be registered within the roles module and referenced by their
+    string values.
+
+    Assumes that the decorated class has a "ROLE" attribute, defining its type.
+
+    """
+    try:
+        role_name = getattr(cls, 'ROLE')
+        REGISTERED_ACCESS_ROLES[role_name] = cls
+    except AttributeError:
+        log.exception(u"Unable to register Access Role with attribute 'ROLE'.")
```

Comment:
Hmm, I thought class decorators had to return the class (just like function decorators need to return a function).